QuestionAnswering/QA_ICL/README.md

```diff
@@ -1,15 +1,199 @@
+    F[Triton inference server] -->|Returns inferences| C
+
+    classDef database fill:#f9f,stroke:#333,stroke-width:2px;
+    class D,E database;
+```
+
+## Key Components
+1. **Frontend**: A Next.js application serving as the user interface.
+2. **Backend**: A FastAPI application handling core logic and data processing.
+    - **Redis**: Used for semantic parsing example retrieval, entity retrieval via fuzzy search, and temporary storage of QA requests for user session management.
+    - **Triton Inference Server**: Manages machine learning model inference.
+3. **Blazegraph Servers**: Store and manage RDF data, accessible via SPARQL protocol.
 
 ## Project Structure
 
-- [`data_generation`](./data_generation): encompases Python scripts to prepare data for entity linking and in-context learning.
-- [`backend`](./backend): Marie's backend
-- [`frontend`](./frontend): Next app for Marie frontend and a minimal mock backend server
+The project is organised into the following main directories:
+
+- `data_generation/`: Python scripts for entity linking and ICL data preparation. The scripts generate three kinds of data:
+  - Lexicon: A collection of ontology representations in texts for classes, predicates, or entities.
+  - Simplified graph schema: Captures node types, edge types, and relation types.
+  - Examples of data request: Pairs of natural language queries and their corresponding data requests for ICL.
+- `backend/`: The backend, called Marie Backend, it contains the FastAPI application serving as the system's backend. It consists of two main components:
+  - `fastapi_app`: A FastAPI application.
+  - `triton_inference_server`: Serves the Sentence-BERT model for generating text embedding.
+- `frontend/`: The frontend, called Marie Frontend, serves as the user interface for interacting with the backend services. Built using Next.js, it communicates with the backend to perform operations such as querying chemical species and filtering zeolites, and displays the results to users in a user-friendly manner.
+
+## Getting Started
+
+### Prerequisities
+ - Docker (recommended for deployment)
+ - Python 3.8 or higher
+ - Node.js and npm (for frontend development)
+
+### Deployment Setup
+1. Clone the TWA repository and go to the QA_ICL folder
+
+    ```
+    cd TheWorldAvatar/QuestionAnswering/QA_ICL/
+    ```
+2. Backend Setup
+   - Follow steps [here](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/QuestionAnswering/QA_ICL/backend#docker-deployment-recommended), which covers the following topics
+      - Populate data for `Fastapi App` and `triton`
+      - Ingest data to `Fastapi App` and `triton`
+      - Launch `Fastapi App` and `triton` containers
+   - Configuration
+     - Fastapp Credentials: create file `app.local.yaml` as instructed [here](https://github.com/cambridge-cares/TheWorldAvatar/tree/main/QuestionAnswering/QA_ICL/backend/fastapi_app#configurable-parameters) and provide credentials
```

Comment:
"Fastapp Credentials" implies that they are the credentials to access the FastAPI app. This is not the case. Consider using the term "App parameters" or "App settings" instead.